libs/service-portal/core/src/utils/formatPlausiblePathToParams.ts

```diff
@@ -0,0 +1,35 @@
+import { ParamType } from '@island.is/plausible'
+import { matchPath } from 'react-router-dom'
+import { ServicePortalPath } from '../lib/navigation/paths'
+/**
+ * Constructing the plausible tracking object.
+ * Service-portal must de-construct all identifiable user data before passing info on-to plausible.
+ * @param path ServicePortalPath of current page
+ * @param fileName optional param for document plausible events
+ * @returns ParamType: {
+ * url: string,
+ * location: string,
+ * fileName: string
+ * }
+ */
+export const formatPlausiblePathToParams = (
+  path: string,
+  fileName?: string,
+) => {
+  const routes = Object.values(ServicePortalPath)
+  const currentPath = matchPath(path, {
+    path: routes,
+    exact: true,
+    strict: true,
+  })?.path
+
+  const pageOrigin = window.location.origin
+  const rootPath = ServicePortalPath.MinarSidurPath
+  const absoluteUrl = `${pageOrigin}${rootPath}${currentPath ?? ''}`
```

Comment:
Isn't possible for `currentPath` to have identifiable info like nationalId? If the route has a `:nationalId` segment? For example the `Child`, `FamilyMember` and `Spouse` routes have it. Wouldn't that result in the nationalId being sent to plausible if they are using events?

Reply:
Yes! That's exactly what's happening today and what I want to prevent with this change..

So here the `currentPath` coming from line 20 will go through matchPath from `react-router` where it will match the current path, /family/123, with the corresponding `ServicePortalPath` /family/:nationalId and return the `ServicePortalPath`.

So this:

`formatPlausiblePathToParams("/minar-sidur/123")`

should return:
```
{
    url: "https://www.islandis/minar-sidur/:nationalId",
    location: "minar-sidur/:nationalId",
  }
  ```

I was sending the wrong path in the location though, I have fixed it and added one more test case.

Reply:
Perfect. I was mixing the `path` and `url` properties from `matchPath` together.
Thanks for the clarification.